message.go

```diff
@@ -84,14 +84,20 @@ func (m *Message) Finish() {
 // One of Finish or Requeue should be called on every message, and the methods
 // will panic if they are called more than once.
 func (m *Message) Requeue(timeout time.Duration) {
-	if m.cmdChan == nil {
+	if m.Done() {
 		panic("(*Message).Finish or (*Message).Requeue has already been called")
 	}
 	defer func() { recover() }() // the connection may have been closed asynchronously
 	m.cmdChan <- Req{MessageID: m.ID, Timeout: timeout}
 	m.cmdChan = nil
 }
 
+// Done will return a bool indicating whether Finish or Requeue has been called
+// for this message.
+func (m *Message) Done() bool {
```

Comment:
Good thinking. I'll use "Complete" since I think "Finished" would sound like it would return `false` if you called `Message.Requeue()`.